lib/util/memcache.ex

```diff
@@ -73,6 +73,31 @@ defmodule Antikythera.Memcache do
     end)
   end
 
+  @doc """
+  Try to read a value associated with the `key` from #{inspect(__MODULE__)} and if that fails, write a value returned by `fun` to #{inspect(__MODULE__)}.
+
+  `fun` is evaluated only if a value is not found, and the new value returned by `fun` is stored in #{inspect(__MODULE__)}.
+  If a value is found in #{inspect(__MODULE__)} or writing the new value to #{inspect(__MODULE__)} succeeds, the value is returned as `{:ok, value}`, but if writing the new value fails, an error is returned in the same manner as `write/5`.
+
+  Parameters `lifetime_in_sec` and `prob_lifetime_ratio` are used to call `write/5` and the details are described above.
+  """
+  defun read_or_else_write(key                 :: Key.t,
+                           epool_id            :: v[EPoolId.t],
+                           lifetime_in_sec     :: v[non_neg_integer],
+                           prob_lifetime_ratio :: v[NormalizedFloat.t] \\ @default_ratio,
+                           fun                 :: (-> Value.t)) :: R.t(Value.t, :too_large_key | :too_large_value) do
+    read(key, epool_id)
+    |> case do
```

Comment:
Minor issue:
Let's collapse the 2 lines into 1`|>` as 1st arg to `case` macro is sufficiently simple.
```ex
case read(key, epool_id) do
```

Reply:
Fixed in f65471aad7c25f1284de55d7d341aa7de52c525f